### PR TITLE
*: add rpc error stats and refine log

### LINKED ISSUE #52809: improve region request log for diagnose and add rpc error stats
## Enhancement

See:
- https://github.com/tikv/client-go/issues/1315
- https://github.com/tikv/client-go/pull/1300

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Issue Number: close #52809

Related PR: https://github.com/tikv/client-go/pull/1300

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -322,3 +322,5 @@ replace (
 	sourcegraph.com/sourcegraph/appdash => github.com/sourcegraph/appdash v0.0.0-20190731080439-ebfcffb1b5c0
 	sourcegraph.com/sourcegraph/appdash-data => github.com/sourcegraph/appdash-data v0.0.0-20151005221446-73f23eafcf67
 )
+
+replace github.com/tikv/client-go/v2 => github.com/crazycs520/client-go/v2 v2.0.0-alpha.0.20240425073930-b36d4a84d7cb
```
Comment (zimulala) [inline go.mod:326]: Why not use "github.com/tikv/client-go" directly?

Comment (crazycs520) [reply go.mod:326]: Since https://github.com/tikv/client-go/pull/1300 isn't merge.

**Review by cfzjywxk — APPROVED** [review]

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**